File: Python/palindromic_string.py
"""
Given a string s check if it is palindrome or not.

Input:
The first line contains 'T' denoting the number of test cases. Then follows description of test cases.
Each case begins with a single integer N denoting the length of string. The next line contains the string s.


Output:
Print "Yes" if it is a palindrome else "No". (Without the double quotes)

Example:
Input:
1
4
abba

Output:
Yes
"""
t = int(input())
while t > 0:
    l = input()
    str = input()
    if str == str[::-1]:
        print("Yes")
    else:
        print("No")
    t -= 1

File: C/hackerearth/palindromic_string.c
/* You have been given a String S.You need to find and print whether this string is a palindrome or not. 
If yes, print "YES" (without quotes), else print "NO" (without quotes).

Input Format: The first and only line of input contains the String S

Output Format: Print the required answer on a single line. 

Sample Input: aba
Sample Output: YES */
#include <stdio.h>
#include <string.h>
int main(){
    char s[100];
    int flag=1,l;
    scanf("%s",s);
    l=strlen(s);
    for(int i=0;i<=l;i++){
        if(s[i]!=s[strlen(s)-i-1]){
            flag=0;
            break;
        }
        l--;
    }
    if(flag==1)
        printf("YES");
    else
        printf("NO");
    return 0;
}